src/geop/main.py

```diff
@@ -66,6 +44,66 @@ def count():
     })
 
 
+@app.route('/pair-counts', methods=['POST'])
+def pair_counts():
+    """
+    For a pair of rasters whose extents and srs match, count cell pairs that
+    occur when the rasters are stacked.  This resembles the geoprocessing
+    required to run TR-55.
+    """
+    user_input = parse_config(request)
+
+    start = time.clock()
+    geom = reproject(
+            shape(user_input['query_polygon']),
+            to_srs=user_input['srs'])
+
+    # Read in two rasters and mask geom on both of them
+    layers = tuple(mask_geom_on_raster(geom, raster_path)
+                   for raster_path in user_input['raster_paths'])
+
+    # Take the two masked arrays, and stack them along the third axis
+    # Effectively: [[cell_1a, cell_1b], [cell_2a, cell_2b], ..],[[...]]
+    pairs = np.ma.dstack(layers)
+
+    # Get the array in 2D form
+    arr = pairs.reshape(-1, pairs.shape[-1])
+
+    # Remove Rows which have masked values
+    trim_arr = np.ma.compress_rowcols(arr, 0)
+
+    # Lexicographically sort so that repeated pairs follow one another
+    sorted_arr = trim_arr[np.lexsort(trim_arr.T), :]
+
+    # Get the indices where a new row of pairs appears
+    diff_indexes = np.where(np.any(np.diff(sorted_arr, axis=0), 1))[0]
```

Comment:
Is `np.diff` causing like cells in both rasters to negate themselves in a way that only produces entries in `diff_indexes` for indexes (after `np.any` and `np.where` are applied) that aren't zeroed out?